test/bucket_test.js

```diff
@@ -293,6 +293,220 @@ describe("Bucket", () => {
     });
   });
 
+  /** @test {Bucket#listGroups} */
+  describe("#listGroups()", () => {
+    const data = [{id: "a"}, {id: "b"}];
+
+    beforeEach(() => {
+      sandbox.stub(client, "execute").returns(Promise.resolve(data));
+    });
+
+    it("should list bucket groups", () => {
+      getBlogBucket().listGroups();
+
+      sinon.assert.calledWithMatch(client.execute, {
+        path: "/buckets/blog/groups"
+      });
+    });
+
+    it("should merge default options", () => {
+      getBlogBucket({headers: {Foo: "Bar"}})
+        .listGroups({headers: {Baz: "Qux"}});
+
+      sinon.assert.calledWithMatch(client.execute, {
+        headers: {Foo: "Bar", Baz: "Qux"}
+      });
+    });
+
+    it("should return the list of groups", () => {
+      return getBlogBucket().listGroups()
+        .should.become(data);
+    });
+  });
+
+  /** @test {Bucket#getGroup} */
+  describe("#getGroup", () => {
+    const fakeGroup = {data: {}, permissions: {}};
+    beforeEach(() => {
+      sandbox.stub(client, "execute").returns(Promise.resolve({
+        json: fakeGroup
+      }));
+    });
+
+    it("should extend request headers with optional ones", () => {
+      getBlogBucket({headers: {Foo: "Bar"}})
+        .getGroup("foo", {headers: {Baz: "Qux"}});
+
+      sinon.assert.calledWithMatch(client.execute, {
+        path: "/buckets/blog/groups/foo",
+        headers: {Foo: "Bar", Baz: "Qux"}
+      });
+    });
+
+    it("should return the group", () => {
+      getBlogBucket().getGroup("foo")
```

Comment:
Wait, I'm surprised this works as mocha expects a promise to be returned in order to perform the eventual assertion. Please double-check this test actually breaks if you change `fakeGroup`.